mantidproject/mantid
# GSASIIRefineFitPeaks output fitted peaks workspace

The fitted peaks from GSAS-II are now output as a workspace from GSASIIRefineFitPeaks. A lot of shared code was also refactored into a common helper class in the system test

**To test:**

**Make sure the GSASIIRefineFitPeaks system test passes locally - this is not currently run on the build server**

You'll need GSAS-II installed. Install it using one of the scripts in `scripts/GSAS-II` or follow the instructions [here](https://subversion.xray.aps.anl.gov/trac/pyGSAS).

Sample data is at `\\olympic\Babylon5\Public\JoeRamsay\GSASIIRefineFitPeaks`.

Run the following script (replace the marked variables with your GSAS-II installation directory and the name of a new GSAS-II project file, which will be created by the algorithm)
```
import os

# Replace this
path_to_gsas = "/path/to/directory/containing/GSASIIscriptable.py"

data_dir = r"\\olympic\Babylon5\Public\JoeRamsay\GSASIIRefineFitPeaks"

ws_path = os.path.join(data_dir, "ENGINX_280625_focused_bank_1.nxs")
spectrum = Load(Filename=ws_path, OutputWorkspace="ENGINX_280625_1")

# Replace this
gsas_proj_file = "/somewhere/local/Test.gpx"

phase_file = os.path.join(data_dir, "FE_ALPHA.cif")
iparams_file = os.path.join(data_dir, "template_ENGINX_241391_236516_North_bank.prm")

GSASIIRefineFitPeaks(PathToGSASII=path_to_gsas, 
                     InputWorkspace=spectrum,
                     SaveGSASIIProjectFile=gsas_proj_file,
                     PhaseInfoFile=phase_file,
                     InstrumentFile=iparams_file,
                     RefinementMethod="Rietveld refinement",
                     OutputWorkspace="FittedPeaks")
```

Overplot `FittedPeaks` on `ENGINX_280625_1`. The fit isn't great, but you should be able to see that some peaks line up, and a sensible workspace is output.

![capture](https://user-images.githubusercontent.com/29916298/35269830-1d44252e-0025-11e8-8357-854dcc7339d9.PNG)


Fixes #21510 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/f0d8082fb486cadbaa3189dacda0be649ee06a8b)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
